deeplinkdispatch-base/src/test/java/com/airbnb/deeplinkdispatch/CharSequenceUtf8Test.kt

```diff
@@ -0,0 +1,94 @@
+
+    @Test(expected = IllegalArgumentException::class) fun testChunkToSmall(){
+        "\u0000".chunkOnModifiedUtf8ByteSize(2)
+    }
+
+    @Test fun testOneNullByte(){
+        val source = "\u0000"
+        val testResult = source.chunkOnModifiedUtf8ByteSize(5)
+        assertEquals(testResult.size, 1)
+        assertEquals(testResult[0].length, 1)
+        assertTrue(compareSourceAndChunked(source,testResult))
+    }
+
+    @Test fun testChunk(){
+        val source = "\u0000\u0000\u0000"
+        val testResult = source.chunkOnModifiedUtf8ByteSize(5)
+        assertEquals(testResult.size, 2)
+        assertEquals(testResult[0].length, 2)
```

Comment:
Would it make more sense to assert that each chunk's byte size is <= the argument provided to `chunkOnModifiedUtf8ByteSize()`? In theory that's what we care about, not the number of characters in each chunk.

Reply:
I was thinking about that but did not want to implement the chunk method in reverse to test the result but I think I do not have to do that, will add some.